### PR TITLE
Speed up binary file read

### PR DESCRIPTION
**Pull request by lshamis**
Test setup:
```
$ truncate -s 10G example_file
```

Before change:
```
$ time bat example_file
───────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: example_file   <BINARY>
───────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

real	0m12.840s
user	0m3.923s
sys	0m8.911s
```
After update
```
$ time bat example_file
───────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: example_file   <BINARY>
───────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

real	0m0.037s
user	0m0.036s
sys	0m0.002s
```

### CODE REVIEW

**Comment by lshamis** [comment]
This should resolve https://github.com/sharkdp/bat/issues/2262

File: src/input.rs [hunk 1]
```diff
@@ -258,7 +258,11 @@ pub(crate) struct InputReader<'a> {
 impl<'a> InputReader<'a> {
     fn new<R: BufRead + 'a>(mut reader: R) -> InputReader<'a> {
         let mut first_line = vec![];
-        reader.read_until(b'\n', &mut first_line).ok();
+        reader
+            .by_ref()
+            .take(1024)
```
Comment (keith-hall) [inline src/input.rs:263]: forgive my ignorance, but does this not mean that if a text file with a first line longer than 1024 chars is read, that first line would get truncated at 1024 chars?

Comment (lshamis) [reply src/input.rs:263]: Only for the purposes of checking whether the file is binary. The check down below:
```
else if content_type != Some(ContentType::BINARY) && first_line.last() != Some(&b'\n') {
    reader.read_until(b'\n', &mut first_line).ok();
}
```
will complete the first line if the content is not binary. If the content is binary, bat wont show it anyway.

Actually, maybe I need to also check for `self.config.show_nonprintable`.

**Comment by lshamis** [comment]
I'm not sure how I broke `no_args_doesnt_break` on some architectures.

**Comment by sharkdp** [comment]
> I'm not sure how I broke `no_args_doesnt_break` on some architectures.

It might be broken on *all* platforms where the test actually runs? Not that it does not run on Windows. And we don't execute unit tests on all platforms...

**Comment by lshamis** [comment]
Sorry, I don't have time right now to debug. I close this PR for now and will try to complete it in the future. Thanks.

**Comment by sharkdp** [comment]
No worries. And no rush. I suggest we simply keep it open. Maybe someone else wants to look into it

**Comment by Enselic** [comment]
I'm closing this PR for now to keep the PR inbox clean.

I have moved/copied the "help wanted" label to the issue itself: https://github.com/sharkdp/bat/issues/2262

Of course feel free to re-open this PR if work resumes on it!

**Pull request closed without merging**